File: applicatie/src/main/java/com/vdzon/administratie/util/ReflectionUtils.java
package com.vdzon.administratie.util;

import java.lang.reflect.InvocationTargetException;
import java.lang.reflect.Method;

public class ReflectionUtils {

    public static Object getInstance(Class clazz) {
        try {
            return  clazz.newInstance();
        } catch (InstantiationException e) {
            e.printStackTrace();
        } catch (IllegalAccessException e) {
            e.printStackTrace();
        }
        throw new RuntimeException("Kon geen instantie maken");
    }

    public static Object callMethod(Method method, Object instance, Object... args) {
        try {
            return method.invoke(instance,args);
        } catch (IllegalAccessException e) {
            e.printStackTrace();
        } catch (InvocationTargetException e) {
            e.printStackTrace();
        }
        throw new RuntimeException("Kon methode niet aanroepen");
    }


}

File: applicatie/src/main/java/com/vdzon/administratie/util/ReflectionUtils.kt
package com.vdzon.administratie.util

import java.lang.reflect.InvocationTargetException
import java.lang.reflect.Method

object ReflectionUtils {

    fun getInstance(clazz: Class<*>): Any {
        try {
            return clazz.newInstance()
        } catch (e: InstantiationException) {
            e.printStackTrace()
        } catch (e: IllegalAccessException) {
            e.printStackTrace()
        }

        throw RuntimeException("Kon geen instantie maken")
    }

    fun callMethod(method: Method, instance: Any, vararg args: Any): Any? {
        try {
            return method.invoke(instance, *args)
        } catch (e: IllegalAccessException) {
            e.printStackTrace()
        } catch (e: InvocationTargetException) {
            e.printStackTrace()
        }

        throw RuntimeException("Kon methode niet aanroepen")
    }


}
